### PR TITLE
Add more runtime implementations of builtins

### PR DESCRIPTION
**Pull request by namannimmo10**

### CODE REVIEW

File: src/runtime/lpython_builtin.py [hunk 1]
```diff
@@ -75,3 +75,18 @@ def bool(x: i32) -> bool:
...
+    """
+    Return the length of the string `s`.
+    """
+    if s == '':
+        return 0
+    count: i32
+    count = 0
+    end: i32
+    end = -1
+    while s[count:end] != '':
```
Comment (namannimmo10) [inline src/runtime/lpython_builtin.py:90]: This raises a codegen error when it hits this line 👇 

https://github.com/lcompilers/lpython/blob/19dd8a71ad4262587ff6e3c80482fdefdde86344/src/libasr/codegen/asr_to_llvm.cpp#L1022

File: src/lpython/semantics/python_comptime_eval.h [hunk 2]
```diff
@@ -207,6 +210,64 @@ struct PythonIntrinsicProcedures {
...
+        ASR::expr_t* arg1 = ASRUtils::expr_value(args[0]);
+        ASR::expr_t* arg2 = ASRUtils::expr_value(args[1]);
+        ASR::ttype_t* arg1_type = ASRUtils::expr_type(arg1);
+        ASR::ttype_t* arg2_type = ASRUtils::expr_type(arg2);
+        ASR::ttype_t *int_type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc, 4, nullptr, 0));
+        ASR::ttype_t *real_type = ASRUtils::TYPE(ASR::make_Real_t(al, loc, 8, nullptr, 0));
+        if (!ASRUtils::check_equal_type(arg1_type, arg2_type)) {
+            throw SemanticError("The arguments to pow() must have the same type.", loc);
+        }
+        if (ASRUtils::is_integer(*arg1_type) || ASRUtils::is_integer(*arg1_type)) {
```
Comment (namannimmo10) [inline src/lpython/semantics/python_comptime_eval.h:253]:
```suggestion
        if (ASRUtils::is_integer(*arg1_type) && ASRUtils::is_integer(*arg2_type)) {
```

File: src/runtime/lpython_builtin.py (hunk 1 above)
Comment (certik) [reply src/runtime/lpython_builtin.py:90]: Right. `len` is a truly "builtin" function. It does not have an implementation using LPython. Implement it like this:
```python
def len(s: str) -> i32:
    """
    Return the length of the string `s`.
    """
    pass
```
The LLVM backend then has to provide its own implementation of `len` (which it already does).

Comment (certik) [reply src/runtime/lpython_builtin.py:90]: Other functions in this category are numpy's `size`, `shape`, etc. They all have to be implemented by the backend directly. Other functions like `chr` can (and should) be implemented in LPython.

File: src/runtime/lpython_builtin.py [hunk 3]
```diff
@@ -75,3 +75,25 @@ def bool(x: i32) -> bool:
...
+        count += 1
+    return count + 1
+
+
+def pow(x: i32, y: i32) -> i32:
+    """
+    Returns x**y.
+    """
+    return (x)**(y)
```
Comment (certik) [inline src/runtime/lpython_builtin.py:99]: I would just write it like this:
```suggestion
    return x**y
```
but it's a matter of style.

**Comment by certik** [comment]
This looks great. Do the change for `len` as I suggested above, and see if this works. If not, I can help fix it.

**Comment by namannimmo10** [comment]
> This looks great. Do the change for `len` as I suggested above, and see if this works. If not, I can help fix it.

I tried using `pass`, but now it gives:
`code generation error: Unsupported len value in ASR`

**Comment by certik** [comment]
Just do the change anyway, that's what has to happen. We might need to modify our AST to ASR to generate an interface for "len", not an implementation. Python doesn't have syntax for declaring an interface, we might perhaps later do it with a decorator.

**Comment by certik** [comment]
Looks good. If there is a bug with `len`, please open up an issue.

**Pull request merged**